FRW/Services/Manager/WalletConnect/WalletConnectHandler.swift

```diff
@@ -13,42 +13,84 @@ import WalletConnectSign
 
 // https://github.com/onflow/flow-evm-gateway?tab=readme-ov-file#evm-gateway-endpoints
 
+
+
 struct WalletConnectHandler {
     // MARK: Internal
+    
+    enum HandlerError: Error {
+        case chainIdMismatch
+    }
 
     func isAllowedSession(sessionProposal: Session.Proposal) -> Bool {
         let namespaces = namespaceTag(sessionProposal: sessionProposal)
         let result = allowNamespaces.filter { namespaces.contains($0) }
         return !result.isEmpty
     }
 
-    func chainReference(sessionProposal: Session.Proposal) -> String? {
-        let handle = current(sessionProposal: sessionProposal)
-        return handle.chainReference(sessionProposal: sessionProposal)
-    }
-
     func sessionInfo(sessionProposal: Session.Proposal) -> SessionInfo {
-        let handle = current(sessionProposal: sessionProposal)
-        let info = handle.sessionInfo(sessionProposal: sessionProposal)
+        let appMetadata = sessionProposal.proposer
+        let requiredNamespaces = sessionProposal.requiredNamespaces
+        let info = SessionInfo(
+            name: appMetadata.name,
+            descriptionText: appMetadata.description,
+            dappURL: appMetadata.url,
+            iconURL: appMetadata.icons.first ?? "",
+            chains: [],
```

Comment:
Those empty fields (chians, methods, pendingRequests, data) can be removed from `SessionInfo`, it didn't see any where else is using it. 
Might be WC SDK changed their data structure.

And the `requiredNamespaces` can be remove as well, it's not being used